scene/main/canvas_item.cpp

```diff
@@ -90,7 +91,10 @@ void CanvasItemMaterial::_update_shader() {
 
 	//must create a shader!
 
-	String code = "shader_type canvas_item;\nrender_mode ";
+	// Add a comment to describe the shader origin (useful when converting to ShaderMaterial).
+	String code = "// NOTE: Shader automatically converted from " VERSION_NAME " " VERSION_FULL_CONFIG "'s CanvasItemMaterial.\n\n";
```

Comment:
vformat may be easier to read instead of adding.

Also Godot has a stringbuilder system.

Reply:
Since this is compile-time concatenation, I think we should leave it as-is as this is the most efficient way to do it. Also, this string doesn't need to be translated, so the placeholder will never move.